File: C++/calculator.cpp
#include <stdio.h>

int main() {

  char operation;
  double x, y;
  printf("Select which operation you would like to perform;\n'+' -> addition\n'-' -> subtraction\n'*' -> multiplication\n'/' -> division\nSelection: ");
  scanf("%c", &operation);
  printf("Enter the first number: ");
  scanf("%lf", &x);
  printf("Enter the second number: ");
  scanf("%lf", &y);

  switch (operation) {
    case '+':
      printf("%.1lf + %.1lf = %.1lf", x, y, x + y);
      break;
    case '-':
      printf("%.1lf - %.1lf = %.1lf", x, y, x - y);
      break;
    case '*':
      printf("%.1lf * %.1lf = %.1lf", x, y, x * y);
      break;
    case '/':
      printf("%.1lf / %.1lf = %.1lf", x, y, x / y);
      break;
    default:
      printf("You have not entered a valid operator. Please run the program again.");
  }

  return 0;
}

File: Python/calculator.py
def calculation():
    operation = input('''
Select which operation you would like to perform:
"+" -> addition
"-" -> subtraction
"*" -> multiplication
"/" -> division
Selection: ''')

    x = int(input('Please enter the first number: '))
    y = int(input('Please enter the second number: '))

    if operation == '+':
        print('{} + {} = '.format(x, y))
        print(x + y)

    elif operation == '-':
        print('{} - {} = '.format(x, y))
        print(x - y)

    elif operation == '*':
        print('{} * {} = '.format(x, y))
        print(x * y)

    elif operation == '/':
        print('{} / {} = '.format(x, y))
        print(x / y)

    else:
        print('You have not typed a valid operator, please run the program again.')

    repeat()

def repeat():
    calc_again = input('''
Do you want perform another calculation?
Type "Y" for YES or "N" for NO.
Y/N: ''')

    if calc_again.upper() == 'Y':
        calculation()
    elif calc_again.upper() == 'N':
        print('Closing the program...')
    else:
        repeat()

calculation()